lib/logproto/logproto-server.c

```diff
@@ -122,12 +122,20 @@ log_proto_server_init(LogProtoServer *self, LogTransport *transport, const LogPr
   self->transport = transport;
 }
 
-void
+gboolean
 log_proto_server_options_set_encoding(LogProtoServerOptions *self, const gchar *encoding)
 {
-  if (self->encoding)
-    g_free(self->encoding);
+  GIConv convert;
+
+  g_free(self->encoding);
   self->encoding = g_strdup(encoding);
+
+  /* validate encoding */
+  convert = g_iconv_open("utf8", encoding);
```

Comment:
I have proposed a recommended improvement in our process that is completely
orthogonal to the bug.

Please move the discussion relating to this bug to its own issue. (My
solution was not making it upper case, we were reading the same code, so
you should know that. And by the way, the problem has been solved (and/or
worked around), so I'm not sure what @lbudai is investigating.) I'd also
thank if we refrained from personal attacks. Comment on the solution, not
on the person.

I still think that we should use atoms whenever feasible. If we move to
UTF-16, should we replace all these strings? I'd actually introduce a layer
for internal conversions instead of these manual calls, so I'm open to all
suggestions.

On Tue, May 10, 2016 at 9:36 AM, Budai Laszlo <notifications@github.com>
wrote:

> In lib/logproto/logproto-server.c
> <https://github.com/balabit/syslog-ng/pull/797#discussion_r62624464>:
>
> >    self->encoding = g_strdup(encoding);
> > +
> > +  /* validate encoding */
> > +  convert = g_iconv_open("utf8", encoding);
>
> It could be a problem... (however, I have a SUTC - small&ugly test code
> and it works when tocode and fromcode is the same...) Maybe different
> library versions...
>
> In libiconv, they convert the character encoding names to upper case, so
> the solution is definitely not what @bkil-syslogng
> <https://github.com/bkil-syslogng> suggested.
> Comment from libiconv we have:
> Before calling aliases_lookup, convert the input string to upper case
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/797/files/35c68349ce999d0318e59abcb3cf4cccef630c81..3fa60288548b54bcfad7b42deb145ef2eafa029c#r62624464>
>